### ISSUE TITLE
Suspiciously large gas overhead

### ISSUE THREAD

**Issue by sirgarfieldc** [libevm/cpo#2]
I am testing this repo out and I am observing around 2.8K gas overhead (difference between calling logic contract directly vs via proxy).

This is significantly larger than the numbers below reported here https://forum.openzeppelin.com/t/a-more-gas-efficient-upgradeable-proxy-by-not-using-storage/4111

Proxy | Gas cost | Gas overhead
-- | -- | --
OpenZeppelin Transparent | 29815 | 2770
Dharma Beacon | 29752 | 2707
EIP-1882 UUPS | 28679 | 1634
Storageless Beacon | 28629 | 1584

I am wondering if there is something that I did wrong.

My compiler setting:

```
Solc 0.8.17
via_ir = true
optimizer = true
optimizer_runs = 2000

```

Can other users share their overhead number ? 

Thanks

**Comment by libevm** [comment]
The post is a few years old, its likely that the Ethereum protocol went through some changes that resulted in opcodes being priced differently

**Comment by sirgarfieldc** [comment]
> The post is a few years old, its likely that the Ethereum protocol went through some changes that resulted in opcodes being priced differently

makes sense.  but can you confirm if the 2.8k number is inline with what you currently observe in your usage ?
